### PR TITLE
Features/linting on pr #9

### PR DESCRIPTION
**Pull request by nfiniteset**
* [x] Create shareable eslint config package
* [x] Set up lint in components package
* [x] Set up lint in react-playground package
* [x] Set up lint in scripts package
* [x] Set up lint in style package
* [x] Get build passing on CI
* [x] Run test & lint on CI
* [x] All tests passing in development
* [x] extend eslint-config-airbnb

### CODE REVIEW

**Comment by nfiniteset** [comment]
Given that each package uses different tech and will likely need _some_ different linting rules I'm taking the approach that each package:
- Declares it's own linting dependencies 
- Implements `npm run lint` to lint it's own code
- `learn run lint` will run on each PR

Does this fit with your assumptions @camwest?

A risk with this approach is that the rules that ought to be shared are duplicated across each package. Perhaps this can be alleviated in some way.

**Comment by camwest** [comment]
what if we make a package called orion-lint and store he base rules there. Then each package can use it and define their own npm run lint. What do you think?

**Comment by nfiniteset** [comment]
I was just looking into that. Seems doable. Was looking at this: http://eslint.org/docs/user-guide/configuring#using-a-shareable-configuration-package

**Comment by camwest** [comment]
Yeah - I hear you about different technologies, I think if our base is 'generic' JS and then we layer in on framework specifics inside the respective repos that would be acceptable

**Comment by nfiniteset** [comment]
Progress update @camwest 

I've got a shared eslint config package set up. The components, react-playground and scripts packages extend it. Seems like a workable approach.

The build is passing again on Circle 😄 but failing in development 😢. Some layer of the testing stack is stubbing style modules and returning the string 'test-file-stub' instead of the module, failing the test. I'm scratching my head on this one.

I'm also getting held up by the style module, which can't seem to find the @orion-ui/eslint-config-orion package.

I could use some fresh eyes on those two issues.

File: packages/style/2016-12-01/index.js [hunk 1]
```diff
@@ -1,6 +1,6 @@
...
-  BorderRadius: require('../modules/border-radius'),
-  Hovers: require('../modules/hovers'),
-  Skins: require('../modules/skins'),
-  Spacing: require('../modules/spacing')
-}
\ No newline at end of file
+  BorderRadius: require('../modules/border-radius.js'),
+  Hovers: require('../modules/hovers.js'),
+  Skins: require('../modules/skins.js'),
+  Spacing: require('../modules/spacing.js')
```
Comment (nfiniteset) [inline packages/style/2016-12-01/index.js]: The module mapper looks for calls to `require` that don't include `js` or `jsx` in the module name. Implicit file extensions get us into trouble here. Odd, huh?

File: packages/eslint-config-orion/index.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        "quotes": [
+            "error",
+            "single"
+        ],
+        "semi": [
+            "error",
+            "always"
+        ]
+    }
+}
```
Comment (nfiniteset) [inline packages/eslint-config-orion/index.js]: I chose these rules arbitrarily. Do you want to do anything more rigorous with rules right now?

**Comment by nfiniteset** [comment]
@camwest Ok. I’ve got it down to a single problem. A test is failing when `private-registry.js` calls `window.customElements`. No window in node… Any insight here?

**Comment by camwest** [comment]
Chatted with @nfiniteset on hangouts about this. We're going to guard inside `private-registry` for now to get the build green.

File: packages/eslint-config-orion/index.js (hunk 2 above)
Comment (camwest) [inline packages/eslint-config-orion/index.js]: Can we base our ruleset on an external existing one and then modify from there? Maybe pull airbnbs?

**Comment by camwest** [comment]
Getting the following error:

```
➜  orion git:(features/linting-on-pr-#9) lerna clean
Lerna v2.0.0-beta.30
About to remove the following directories:
- /Users/camwest/src/orion-ui/orion/packages/components/node_modules
- /Users/camwest/src/orion-ui/orion/packages/eslint-config-orion/node_modules
- /Users/camwest/src/orion-ui/orion/packages/react-playground/node_modules
- /Users/camwest/src/orion-ui/orion/packages/scripts/node_modules
- /Users/camwest/src/orion-ui/orion/packages/style/node_modules
? Proceed? Yes
All clean!
➜  orion git:(features/linting-on-pr-#9) lerna bootstrap
Lerna v2.0.0-beta.30
Bootstrapping 5 packages
Installing external dependencies
Symlinking packages and binaries
Prepublishing packages
Successfully bootstrapped 5 packages.
➜  orion git:(features/linting-on-pr-#9) npm test

> orion@1.0.0 test /Users/camwest/src/orion-ui/orion
> lerna run test; lerna run lint

Lerna v2.0.0-beta.30

> react-playground@0.1.0 test /Users/camwest/src/orion-ui/orion/packages/react-playground
> CI=true react-scripts test --env=jsdom


Successfully ran npm script 'test' in packages:
- react-playground
Lerna v2.0.0-beta.30

> @orion-ui/scripts@1.0.0 lint /Users/camwest/src/orion-ui/orion/packages/scripts
> eslint .



> @orion-ui/style@1.0.0 lint /Users/camwest/src/orion-ui/orion/packages/style
> eslint .

Cannot find module 'eslint-config-airbnb'
Referenced from: @orion-ui/eslint-config-orion
Referenced from: /Users/camwest/src/orion-ui/orion/packages/style/.eslintrc.json
Error: Cannot find module 'eslint-config-airbnb'
Referenced from: @orion-ui/eslint-config-orion
Referenced from: /Users/camwest/src/orion-ui/orion/packages/style/.eslintrc.json
    at Object.resolve (/Users/camwest/src/orion-ui/orion/packages/style/node_modules/eslint/lib/util/module-resolver.js:74:19)
    at resolve (/Users/camwest/src/orion-ui/orion/packages/style/node_modules/eslint/lib/config/config-file.js:478:33)
    at load (/Users/camwest/src/orion-ui/orion/packages/style/node_modules/eslint/lib/config/config-file.js:495:26)
    at configExtends.reduceRight.e (/Users/camwest/src/orion-ui/orion/packages/style/node_modules/eslint/lib/config/config-file.js:391:36)
    at Array.reduceRight (native)
    at applyExtends (/Users/camwest/src/orion-ui/orion/packages/style/node_modules/eslint/lib/config/config-file.js:362:28)
    at load (/Users/camwest/src/orion-ui/orion/packages/style/node_modules/eslint/lib/config/config-file.js:529:22)
    at configExtends.reduceRight.e (/Users/camwest/src/orion-ui/orion/packages/style/node_modules/eslint/lib/config/config-file.js:391:36)
    at Array.reduceRight (native)
    at applyExtends (/Users/camwest/src/orion-ui/orion/packages/style/node_modules/eslint/lib/config/config-file.js:362:28)

Errored while running npm script 'lint' in '@orion-ui/style'
Error: Command failed: npm run lint

npm ERR! Darwin 16.1.0
npm ERR! argv "/Users/camwest/.nvm/versions/node/v6.9.1/bin/node" "/Users/camwest/.nvm/versions/node/v6.9.1/bin/npm" "run" "lint"
npm ERR! node v6.9.1
npm ERR! npm  v3.10.8
npm ERR! code ELIFECYCLE
npm ERR! @orion-ui/style@1.0.0 lint: `eslint .`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the @orion-ui/style@1.0.0 lint script 'eslint .'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the @orion-ui/style package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     eslint .
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs @orion-ui/style
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls @orion-ui/style
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /Users/camwest/src/orion-ui/orion/packages/style/npm-debug.log

    at ChildProcess.exithandler (child_process.js:206:12)
    at emitTwo (events.js:106:13)
    at ChildProcess.emit (events.js:191:7)
    at maybeClose (internal/child_process.js:877:16)
    at Socket.<anonymous> (internal/child_process.js:334:11)
    at emitOne (events.js:96:13)
    at Socket.emit (events.js:188:7)
    at Pipe._handle.close [as _onclose] (net.js:498:12)
Errored while running RunCommand.execute
Error: Command failed: npm run lint

npm ERR! Darwin 16.1.0
npm ERR! argv "/Users/camwest/.nvm/versions/node/v6.9.1/bin/node" "/Users/camwest/.nvm/versions/node/v6.9.1/bin/npm" "run" "lint"
npm ERR! node v6.9.1
npm ERR! npm  v3.10.8
npm ERR! code ELIFECYCLE
npm ERR! @orion-ui/style@1.0.0 lint: `eslint .`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the @orion-ui/style@1.0.0 lint script 'eslint .'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the @orion-ui/style package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     eslint .
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs @orion-ui/style
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls @orion-ui/style
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /Users/camwest/src/orion-ui/orion/packages/style/npm-debug.log

    at ChildProcess.exithandler (child_process.js:206:12)
    at emitTwo (events.js:106:13)
    at ChildProcess.emit (events.js:191:7)
    at maybeClose (internal/child_process.js:877:16)
    at Socket.<anonymous> (internal/child_process.js:334:11)
    at emitOne (events.js:96:13)
    at Socket.emit (events.js:188:7)
    at Pipe._handle.close [as _onclose] (net.js:498:12)
Waiting for 2 child processes to exit. CTRL-C to exit immediately.

> @orion-ui/components@1.0.0 lint /Users/camwest/src/orion-ui/orion/packages/components
> eslint .

Cannot find module 'eslint-config-airbnb'
Referenced from: @orion-ui/eslint-config-orion
Referenced from: /Users/camwest/src/orion-ui/orion/packages/components/.eslintrc.json
Error: Cannot find module 'eslint-config-airbnb'
Referenced from: @orion-ui/eslint-config-orion
Referenced from: /Users/camwest/src/orion-ui/orion/packages/components/.eslintrc.json
    at Object.resolve (/Users/camwest/src/orion-ui/orion/packages/components/node_modules/eslint/lib/util/module-resolver.js:74:19)
    at resolve (/Users/camwest/src/orion-ui/orion/packages/components/node_modules/eslint/lib/config/config-file.js:478:33)
    at load (/Users/camwest/src/orion-ui/orion/packages/components/node_modules/eslint/lib/config/config-file.js:495:26)
    at configExtends.reduceRight.e (/Users/camwest/src/orion-ui/orion/packages/components/node_modules/eslint/lib/config/config-file.js:391:36)
    at Array.reduceRight (native)
    at applyExtends (/Users/camwest/src/orion-ui/orion/packages/components/node_modules/eslint/lib/config/config-file.js:362:28)
    at load (/Users/camwest/src/orion-ui/orion/packages/components/node_modules/eslint/lib/config/config-file.js:529:22)
    at configExtends.reduceRight.e (/Users/camwest/src/orion-ui/orion/packages/components/node_modules/eslint/lib/config/config-file.js:391:36)
    at Array.reduceRight (native)
    at applyExtends (/Users/camwest/src/orion-ui/orion/packages/components/node_modules/eslint/lib/config/config-file.js:362:28)

Errored while running npm script 'lint' in '@orion-ui/components'
Error: Command failed: npm run lint

npm ERR! Darwin 16.1.0
npm ERR! argv "/Users/camwest/.nvm/versions/node/v6.9.1/bin/node" "/Users/camwest/.nvm/versions/node/v6.9.1/bin/npm" "run" "lint"
npm ERR! node v6.9.1
npm ERR! npm  v3.10.8
npm ERR! code ELIFECYCLE
npm ERR! @orion-ui/components@1.0.0 lint: `eslint .`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the @orion-ui/components@1.0.0 lint script 'eslint .'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the @orion-ui/components package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     eslint .
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs @orion-ui/components
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls @orion-ui/components
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /Users/camwest/src/orion-ui/orion/packages/components/npm-debug.log

    at ChildProcess.exithandler (child_process.js:206:12)
    at emitTwo (events.js:106:13)
    at ChildProcess.emit (events.js:191:7)
    at maybeClose (internal/child_process.js:877:16)
    at Socket.<anonymous> (internal/child_process.js:334:11)
    at emitOne (events.js:96:13)
    at Socket.emit (events.js:188:7)
    at Pipe._handle.close [as _onclose] (net.js:498:12)
npm ERR! Test failed.  See above for more details.
```

**Pull request merged**